src/ParallelAlgorithms/Actions/MemoryMonitor/ProcessGroups.hpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/MemoryMonitor/MemoryMonitor.hpp"
+#include "Parallel/Serialize.hpp"
+#include "Parallel/TypeTraits.hpp"
+
+namespace mem_monitor {
+/*!
+ * \brief Simple action meant to be run on every branch of a Nodegroup that
+ * computes the size of the local branch and reports that size to the
+ * MemoryMonitor using the ContributeMemoryData simple action.
+ */
+struct ProcessNodegroup {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/, const double time) {
+    static_assert(
+        Parallel::is_nodegroup_v<ParallelComponent>,
+        "ProcessNodegroup can only be run on a Nodegroup parallel component.");
+
+    auto& singleton_proxy =
+        Parallel::get_parallel_component<MemoryMonitor<Metavariables>>(cache);
+
+    auto& nodegroup_proxy =
+        Parallel::get_parallel_component<ParallelComponent>(cache);
+    const int my_node =
```

Comment:
isn't this just `array_index` (and thus can't we just have one action that works on both groups and nodegroups)?

Reply:
You're right. `ContribtueMemoryData` expects an `int` for the node/proc so I just `static_assert`ed that the array index was an `int` inside `ProcessGroups`. The only component that doesn't have `int` for the array index is `Array` (and you shouldn't be running this action on an `Array`) so we would probably have been good without it, but I added it just in case.